kernel/sched.c

```diff
@@ -1470,6 +1472,9 @@ static int32_t z_tick_sleep(k_ticks_t ticks)
 	if (ticks > 0) {
 		return ticks;
 	}
+#else
+	while(expected_wakeup_ticks > sys_clock_tick_get_32())
```

Comment:
It'd be good to have a small comment in the code explaining why this is need here.